File: server/internal/device/device.go
package device

import "time"

// Device is the JSON shape we expose to the UI. It is not the OneStep GPS payload.
type Device struct {
	ID          string     `json:"id"`
	Name        string     `json:"name"`
	Make        string     `json:"make"`
	Model       string     `json:"model"`
	ActiveState string     `json:"active_state"`
	Online      bool       `json:"online"`
	DriveStatus string     `json:"drive_status"`
	Latitude    *float64   `json:"latitude"`
	Longitude   *float64   `json:"longitude"`
	Heading     float64    `json:"heading"`
	SpeedKmh    float64    `json:"speed_kmh"`
	LastSeenAt  *time.Time `json:"last_seen_at"`
}

File: ui/src/types/device.ts
export type Device = {
  id: string
  name: string
  make: string
  model: string
  active_state: string
  online: boolean
  drive_status: string
  latitude: number | null
  longitude: number | null
  heading: number
  speed_kmh: number
  last_seen_at: string | null
}

export type DeviceApi = {
  listDevices: () => Promise<Device[]>
}
